toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -293,44 +295,108 @@ func isSpecialPartition(fstabEntry diskutils.FstabEntry) bool {
 	}
 }
 
-func findSourcePartition(source string, partitions []diskutils.PartitionInfo) (string, error) {
-	_, partition, _, err := findSourcePartitionHelper(source, partitions)
+func findSourcePartition(source string, partitions []diskutils.PartitionInfo,
+	buildDir string,
+) (string, error) {
+	_, partition, _, err := findSourcePartitionHelper(source, partitions, buildDir)
 	if err != nil {
 		return "", err
 	}
 
 	return partition.Path, nil
 }
 
-func findSourcePartitionHelper(source string,
-	partitions []diskutils.PartitionInfo,
+func findSourcePartitionHelper(source string, partitions []diskutils.PartitionInfo,
+	buildDir string,
 ) (imagecustomizerapi.MountIdentifierType, diskutils.PartitionInfo, int, error) {
```

Comment:
Looking at the users of the `findSourcePartitionHelper` function, I think you can change this function to return `ExtendedMountIdentifierType` instead of `MountIdentifierType`. That way, you don't need to do the conversion.